internal/controllers/machineset/machineset_controller_test.go

```diff
@@ -1036,6 +1036,42 @@ func TestMachineSetReconciler_syncMachines(t *testing.T) {
 		},
 	}
 
+	infraMachine := &unstructured.Unstructured{
+		Object: map[string]interface{}{
+			"kind":       "GenericInfrastructureMachine",
+			"apiVersion": "infrastructure.cluster.x-k8s.io/v1beta1",
+			"metadata": map[string]interface{}{
+				"name":      "infra-machine-1",
+				"namespace": namespace.Name,
+				"labels": map[string]interface{}{
+					"machine-set-matching-label": "false", // this label should be modified
+				},
+				"annotations": map[string]interface{}{
+					"annotation-1": "false", // this annotation should be modified
+				},
```

Comment:
nit: Let's also add a few fields that should stay the same (same for BootstrapConfig)